tests/workflow_activate_disable_duplicate_test.php

```diff
@@ -74,7 +74,22 @@ public function test_activate_third() {
      * Test to deactivate the first workflow.
      */
     public function test_deactivate_first() {
-        // TODO to be implemented.
+        workflow_manager::handle_action(action::WORKFLOW_ABORTDISABLE, $this->workflow1->id);
+        $this->assertFalse(workflow_manager::is_active($this->workflow1->id));
+    }
+
+    /**
+     * Test to duplicate the first workflow.
+     */
+    public function test_duplicate_first() {
+        workflow_manager::handle_action(action::WORKFLOW_DUPLICATE, $this->workflow1->id);
+        $workflows = workflow_manager::get_workflows();
+        $this->assertCount(4, $workflows);
+        $duplicate = array_pop($workflows);
```

Comment:
I believe that the element that you retrieve here is not necessarily the duplicate (but it can be, at least in some cases).